recipes-extended/jailhouse/jailhouse/0001-scripts-tools-switch-to-python3.patch

```diff
@@ -0,0 +1,83 @@
+From c9fc6bbd535d0d92578cf3258c6db052ec6094dd Mon Sep 17 00:00:00 2001
+From: Cristinel Panfir <cristinel.panfir@nxp.com>
+Date: Tue, 28 Apr 2020 13:59:21 +0300
+Subject: [PATCH] scripts/tools: switch to python3
+
+Upstream-Status: Pending
+
+Signed-off-by: Cristinel Panfir <cristinel.panfir@nxp.com>
+---
+ scripts/arm64-parsedump.py     | 2 +-
+ scripts/include.mk             | 2 +-
+ tools/jailhouse-cell-linux     | 2 +-
+ tools/jailhouse-cell-stats     | 2 +-
+ tools/jailhouse-config-create  | 2 +-
+ tools/jailhouse-hardware-check | 2 +-
+ 6 files changed, 6 insertions(+), 6 deletions(-)
+
+diff --git a/scripts/arm64-parsedump.py b/scripts/arm64-parsedump.py
+index 9519eb5..c695706 100755
+--- a/scripts/arm64-parsedump.py
++++ b/scripts/arm64-parsedump.py
+@@ -1,4 +1,4 @@
+-#!/usr/bin/env python
++#!/usr/bin/env python3
+ 
+ # Jailhouse, a Linux-based partitioning hypervisor
+ #
+diff --git a/scripts/include.mk b/scripts/include.mk
+index 028f0ab..a8c94b6 100644
+--- a/scripts/include.mk
++++ b/scripts/include.mk
+@@ -42,7 +42,7 @@ INSTALL_PROGRAM ?= $(INSTALL)
+ INSTALL_DATA    ?= $(INSTALL) -m 644
+ INSTALL_DIR     ?= $(INSTALL) -d -m 755
+ 
+-PYTHON ?= python
++PYTHON = /usr/bin/env python3
```

Comment:
Is it OK to re-define conditional assignment here? If yes - then why was that in upstream, and would upstream guys accept the conditional assignment drop?

My concern is: there might be some further discussions with upstream here about this patch, since it still has `Pending` state.

Reply:
I would have liked to get rid of this patch using instead that symlink in the recipe, so as not to mix the python's path:
`ln -sf ${PYTHON} ${STAGING_BINDIR_NATIVE}/python`
I had to put conditional python3 in the patch to make it work.

Reply:
IMHO, modifying source in recipe is not all that great... So symlinks would not be a solution here, I guess.

What I meant is: could you leave the conditional assignment in the patch as it was done in the upstream, something like this:

```
-PYTHON ?= python
+PYTHON ?= /usr/bin/env python3
```

Just replace the interpreter, and leave the assignment as-is. That would get less questions from the upstream guys once the patch is reported.

Reply:
I know what you meant (I wrote it wrong) ..but it doesn't work, it's messing up the python path. 
Maybe I should add an assignment into the recipe